pkgs/by-name/do/dorion/package.nix

```
@@ -1,65 +1,168 @@
+    ./cargo-toml.patch
+  ];
+
+  postPatch = ''
+    # apply cargo-lock patch (we have to do it from src-tauri subdir so cant use patches = [..])
+    (cd /build/source/src-tauri && patch -p1 < ${./cargo-lock.patch})
+
+    # patch cargo-deps
+    (cd $cargoDepsCopy/tauri-plugin-shell-* && patch -p1 < /build/source/src-tauri/patches/tauri-plugin-shell+2.0.1.patch)
+    (cd $cargoDepsCopy/tauri-utils-* && patch -p3 <${./tauri-env-resource-dir.patch})
+    substituteInPlace $cargoDepsCopy/libappindicator-sys-*/src/lib.rs \
+      --replace "libayatana-appindicator3.so.1" "${libayatana-appindicator}/lib/libayatana-appindicator3.so.1"
 
-    mkdir -pv $out
-    mv -v {bin,lib,share} $out
+    # disable pre-build script and disable auto-updater
+    yq -iPo=json '
+        .bundle.resources = (.bundle.resources | map(select(. != "updater*")))
+    ' /build/source/src-tauri/tauri.conf.json
 
-    runHook postInstall
+    # copy shelter injection
+    cp ${shelter} /build/source/src-tauri/injection/shelter.js
+
+    # Very hacky and 99% sure its not even right
+    mkdir -p /build/source/src-tauri/html
+    cp -r /build/source/src/* /build/source/src-tauri/html
+  '';
+
+  preBuild = ''
+    pnpm build:js
   '';
 
+  postInstall = ''
+    mkdir -p $out/lib/dorion
+    cp -r /build/source/src-tauri/{injection,html} $out/lib/dorion
+    desktop-file-edit \
+      --set-comment "Tiny alternative Discord client" \
+      --set-key="Exec" --set-value="Dorion %U" \
+      --set-key="TryExec" --set-value="Dorion" \
+      --set-key="StartupWMClass" --set-value="Dorion" \
+      --set-key="StartupNotify" --set-value="true" \
+      --set-key="Categories" --set-value="Network;InstantMessaging;Chat;" \
+      --set-key="Keywords" --set-value="dorion;discord;vencord;chat;im;vc;ds;dc;dsc;tauri;" \
+      --set-key="Terminal" --set-value="false" \
+      --set-key="MimeType" --set-value="x-scheme-handler/discord" \
+      $out/share/applications/dorion.desktop
+  '';
+
+  env = {
+    TAURI_RESOURCE_DIR = "${placeholder "out"}/lib";
+  };
+
   meta = {
-    homepage = "https://github.com/SpikeHD/Dorion";
+    homepage = "https://spikehd.github.io/projects/dorion/";
     description = "Tiny alternative Discord client";
+    longDescription = ''
+      Dorion is an alternative Discord client aimed towards lower-spec or
+      storage-sensitive PCs that supports themes, plugins, and more!
+    '';
+    changelog = "https://github.com/SpikeHD/Dorion/releases/tag/v${finalAttrs.version}";
+    downloadPage = "https://github.com/SpikeHD/Dorion/releases/tag/v${finalAttrs.version}";
     license = lib.licenses.gpl3Only;
-    mainProgram = "dorion";
-    maintainers = with lib.maintainers; [ aleksana ];
-    platforms = lib.intersectLists (lib.platforms.linux) (lib.platforms.x86_64);
-    sourceProvenance = with lib.sourceTypes; [ binaryNativeCode ];
+    mainProgram = "Dorion";
+    maintainers = with lib.maintainers; [
+      nyabinary
+      aleksana
+      griffi-gh
```

```suggestion
      griffi-gh
      getchoo
```
Feel free to add me as well. I like this as an edge case for cargo-tauri.hook